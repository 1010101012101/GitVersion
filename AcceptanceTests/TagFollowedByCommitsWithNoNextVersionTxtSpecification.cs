using GitHubFlowVersion.AcceptanceTests.Helpers;
using Xunit;

namespace GitHubFlowVersion.AcceptanceTests
{
    using System;
    using ApprovalTests;
    using TestStack.BDDfy;

    public class TagFollowedByCommitsWithNoNextVersionTxtSpecification : IUseFixture<RepositoryFixture>
    {
        private ExecutionResults _result;
        RepositoryFixture _data;

        public void GivenARepositoryWithASingleTagFollowedByCommits()
        {
            _data.Repository.MakeATaggedCommit("0.1.0");
            _data.Repository.MakeACommit();
        }

        public void AndGivenThereIsNoNextVersionTxtFile() {}

        public void WhenGitHubFlowVersionIsExecuted()
        {
            _result = GitVersionHelper.ExecuteIn(_data.RepositoryPath);
        }

        public void ThenNoErrorShouldOccur()
        {
            _result.AssertExitedSuccessfully();
        }

        public void AndTheCorrectVersionShouldBeOutput()
        {
            Approvals.Verify(_result.Output, Scrubbers.GuidScrubber);
        }

        [Fact]
        public virtual void RunSpecification()
        {
            // If we are actually running in teamcity, lets delete this environmental variable
            Environment.SetEnvironmentVariable("TEAMCITY_VERSION", null);
            this.BDDfy();
        }

        public void SetFixture(RepositoryFixture data)
        {
            _data = data;
        }
    }
}
