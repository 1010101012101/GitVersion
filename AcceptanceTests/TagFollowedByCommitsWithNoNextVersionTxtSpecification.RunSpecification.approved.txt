{
  "Major":0,
  "Minor":1,
  "Patch":1,
  "Suffix":"",
  "LongVersion":"0.1.1 Sha:'000000000000000000000000000000000000000'",
  "NugetVersion":"0.1.1",
  "ShortVersion":"0.1.1",
  "BranchName":"master",
  "BranchType":"Master",
  "Sha":"000000000000000000000000000000000000000",
  "MajorMinorPatch":"0.1.1",
  "SemVer":"0.1.1+1",
  "Version":"0.1.1+1",
  "PreReleasePartTwo":1,
  "Stability":"Final"
}
